src/ast/nodes/Identifier.ts

```diff
@@ -152,6 +148,9 @@ export default class Identifier extends NodeBase implements PatternNode {
 
 	hasEffects(context: HasEffectsContext): boolean {
 		if (!this.deoptimized) this.applyDeoptimizations();
+		if (this.variable && (this.variable.kind === 'using' || this.variable.kind === 'await using')) {
+			return true;
```

Comment:
I do not think we need to make every access a side effect. As I understand the spec, the variable is mostly what you expect, i.e. the value is what is assigned in the declarator. However, instead I think we should just ensure that a VariableDeclaration is automatically a side effect if it is of the "using" kind, and is therefore never removed. Or am I overlooking something?

Reply:
Yeah, Detecting `using` in VariableDeclaration is more reasonable and better. Done it!